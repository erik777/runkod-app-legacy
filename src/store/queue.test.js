import reducer, {
  setQueueAct,
  fileOkAct,
  fileErrorAct,
  fileConflictAct,
  conflictFlagAct,
  fileSkipAct
} from "./queue"

let state = undefined;

it('1- default state', () => {
  expect(reducer(state, {})).toMatchSnapshot();
});

it('2- add files to queue. 12 items in "files".', () => {
  const files = [
    {name: 'style.css'}, {name: 'main.js'}, {name: 'vendor.js'},
    {name: 'index.html'}, {name: 'manifest.json'}, {name: 'favicon.ico'},
    {name: 'bg.jpg'}, {name: 'logo.png'}, {name: 'foo.gif'},
    {name: 'bar.png'}, {name: 'baz.js'}, {name: 'lorem.css'}
  ];
  const act = setQueueAct(files);
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('3- file ok action. 11 items in "files", 1 item in "completed".', () => {
  const act = fileOkAct();
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('4- file ok action. 10 item in "files", 2 items in "completed".', () => {
  const act = fileOkAct();
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('5- file ok action. 9 item in "files", 3 items in "completed".', () => {
  const act = fileOkAct();
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});


it('6- file error action. 8 item in "files", 3 items in "completed", 1 item in "failed". ', () => {
  const act = fileErrorAct();
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('7- file conflict action.', () => {
  const act = fileConflictAct();
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});


it('8- conflict flag action.', () => {
  const act = conflictFlagAct(2);
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('9- file ok action. Conflict flag should change to 0.', () => {
  const act = fileOkAct();
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});


it('10- file conflict action.', () => {
  const act = fileConflictAct();
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('11- conflict flag action.', () => {
  const act = conflictFlagAct(5);
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('12- file skip action. Should add an item under "skipped".conflictFlag should be still 5', () => {
  const act = fileSkipAct();
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('13- file skip action. Should add an item under "skipped".conflictFlag should be still 5', () => {
  const act = fileSkipAct();
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});