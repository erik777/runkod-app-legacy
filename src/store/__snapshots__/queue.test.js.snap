// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1- default state 1`] = `
Object {
  "completed": Array [],
  "conflict": false,
  "conflictFlag": 0,
  "current": null,
  "failed": Array [],
  "files": Array [],
  "skipped": Array [],
}
`;

exports[`2- add files to queue. 12 items in "files". 1`] = `
Object {
  "completed": Array [],
  "conflict": false,
  "conflictFlag": 0,
  "current": null,
  "failed": Array [],
  "files": Array [
    Object {
      "name": "style.css",
    },
    Object {
      "name": "main.js",
    },
    Object {
      "name": "vendor.js",
    },
    Object {
      "name": "index.html",
    },
    Object {
      "name": "manifest.json",
    },
    Object {
      "name": "favicon.ico",
    },
    Object {
      "name": "bg.jpg",
    },
    Object {
      "name": "logo.png",
    },
    Object {
      "name": "foo.gif",
    },
    Object {
      "name": "bar.png",
    },
    Object {
      "name": "baz.js",
    },
    Object {
      "name": "lorem.css",
    },
  ],
  "skipped": Array [],
}
`;

exports[`3- file ok action. 11 items in "files", 1 item in "completed". 1`] = `
Object {
  "completed": Array [
    Object {
      "name": "style.css",
    },
  ],
  "conflict": false,
  "conflictFlag": 0,
  "current": null,
  "failed": Array [],
  "files": Array [
    Object {
      "name": "main.js",
    },
    Object {
      "name": "vendor.js",
    },
    Object {
      "name": "index.html",
    },
    Object {
      "name": "manifest.json",
    },
    Object {
      "name": "favicon.ico",
    },
    Object {
      "name": "bg.jpg",
    },
    Object {
      "name": "logo.png",
    },
    Object {
      "name": "foo.gif",
    },
    Object {
      "name": "bar.png",
    },
    Object {
      "name": "baz.js",
    },
    Object {
      "name": "lorem.css",
    },
  ],
  "skipped": Array [],
}
`;

exports[`4- file ok action. 10 item in "files", 2 items in "completed". 1`] = `
Object {
  "completed": Array [
    Object {
      "name": "style.css",
    },
    Object {
      "name": "main.js",
    },
  ],
  "conflict": false,
  "conflictFlag": 0,
  "current": null,
  "failed": Array [],
  "files": Array [
    Object {
      "name": "vendor.js",
    },
    Object {
      "name": "index.html",
    },
    Object {
      "name": "manifest.json",
    },
    Object {
      "name": "favicon.ico",
    },
    Object {
      "name": "bg.jpg",
    },
    Object {
      "name": "logo.png",
    },
    Object {
      "name": "foo.gif",
    },
    Object {
      "name": "bar.png",
    },
    Object {
      "name": "baz.js",
    },
    Object {
      "name": "lorem.css",
    },
  ],
  "skipped": Array [],
}
`;

exports[`5- file ok action. 9 item in "files", 3 items in "completed". 1`] = `
Object {
  "completed": Array [
    Object {
      "name": "style.css",
    },
    Object {
      "name": "main.js",
    },
    Object {
      "name": "vendor.js",
    },
  ],
  "conflict": false,
  "conflictFlag": 0,
  "current": null,
  "failed": Array [],
  "files": Array [
    Object {
      "name": "index.html",
    },
    Object {
      "name": "manifest.json",
    },
    Object {
      "name": "favicon.ico",
    },
    Object {
      "name": "bg.jpg",
    },
    Object {
      "name": "logo.png",
    },
    Object {
      "name": "foo.gif",
    },
    Object {
      "name": "bar.png",
    },
    Object {
      "name": "baz.js",
    },
    Object {
      "name": "lorem.css",
    },
  ],
  "skipped": Array [],
}
`;

exports[`6- file error action. 8 item in "files", 3 items in "completed", 1 item in "failed".  1`] = `
Object {
  "completed": Array [
    Object {
      "name": "style.css",
    },
    Object {
      "name": "main.js",
    },
    Object {
      "name": "vendor.js",
    },
  ],
  "conflict": false,
  "conflictFlag": 0,
  "current": null,
  "failed": Array [
    Object {
      "name": "index.html",
    },
  ],
  "files": Array [
    Object {
      "name": "manifest.json",
    },
    Object {
      "name": "favicon.ico",
    },
    Object {
      "name": "bg.jpg",
    },
    Object {
      "name": "logo.png",
    },
    Object {
      "name": "foo.gif",
    },
    Object {
      "name": "bar.png",
    },
    Object {
      "name": "baz.js",
    },
    Object {
      "name": "lorem.css",
    },
  ],
  "skipped": Array [],
}
`;

exports[`7- file conflict action. 1`] = `
Object {
  "completed": Array [
    Object {
      "name": "style.css",
    },
    Object {
      "name": "main.js",
    },
    Object {
      "name": "vendor.js",
    },
  ],
  "conflict": true,
  "conflictFlag": 0,
  "current": null,
  "failed": Array [
    Object {
      "name": "index.html",
    },
  ],
  "files": Array [
    Object {
      "name": "manifest.json",
    },
    Object {
      "name": "favicon.ico",
    },
    Object {
      "name": "bg.jpg",
    },
    Object {
      "name": "logo.png",
    },
    Object {
      "name": "foo.gif",
    },
    Object {
      "name": "bar.png",
    },
    Object {
      "name": "baz.js",
    },
    Object {
      "name": "lorem.css",
    },
  ],
  "skipped": Array [],
}
`;

exports[`8- conflict flag action. 1`] = `
Object {
  "completed": Array [
    Object {
      "name": "style.css",
    },
    Object {
      "name": "main.js",
    },
    Object {
      "name": "vendor.js",
    },
  ],
  "conflict": false,
  "conflictFlag": 2,
  "current": null,
  "failed": Array [
    Object {
      "name": "index.html",
    },
  ],
  "files": Array [
    Object {
      "name": "manifest.json",
    },
    Object {
      "name": "favicon.ico",
    },
    Object {
      "name": "bg.jpg",
    },
    Object {
      "name": "logo.png",
    },
    Object {
      "name": "foo.gif",
    },
    Object {
      "name": "bar.png",
    },
    Object {
      "name": "baz.js",
    },
    Object {
      "name": "lorem.css",
    },
  ],
  "skipped": Array [],
}
`;

exports[`9- file ok action. Conflict flag should change to 0. 1`] = `
Object {
  "completed": Array [
    Object {
      "name": "style.css",
    },
    Object {
      "name": "main.js",
    },
    Object {
      "name": "vendor.js",
    },
    Object {
      "name": "manifest.json",
    },
  ],
  "conflict": false,
  "conflictFlag": 0,
  "current": null,
  "failed": Array [
    Object {
      "name": "index.html",
    },
  ],
  "files": Array [
    Object {
      "name": "favicon.ico",
    },
    Object {
      "name": "bg.jpg",
    },
    Object {
      "name": "logo.png",
    },
    Object {
      "name": "foo.gif",
    },
    Object {
      "name": "bar.png",
    },
    Object {
      "name": "baz.js",
    },
    Object {
      "name": "lorem.css",
    },
  ],
  "skipped": Array [],
}
`;

exports[`10- file conflict action. 1`] = `
Object {
  "completed": Array [
    Object {
      "name": "style.css",
    },
    Object {
      "name": "main.js",
    },
    Object {
      "name": "vendor.js",
    },
    Object {
      "name": "manifest.json",
    },
  ],
  "conflict": true,
  "conflictFlag": 0,
  "current": null,
  "failed": Array [
    Object {
      "name": "index.html",
    },
  ],
  "files": Array [
    Object {
      "name": "favicon.ico",
    },
    Object {
      "name": "bg.jpg",
    },
    Object {
      "name": "logo.png",
    },
    Object {
      "name": "foo.gif",
    },
    Object {
      "name": "bar.png",
    },
    Object {
      "name": "baz.js",
    },
    Object {
      "name": "lorem.css",
    },
  ],
  "skipped": Array [],
}
`;

exports[`11- conflict flag action. 1`] = `
Object {
  "completed": Array [
    Object {
      "name": "style.css",
    },
    Object {
      "name": "main.js",
    },
    Object {
      "name": "vendor.js",
    },
    Object {
      "name": "manifest.json",
    },
  ],
  "conflict": false,
  "conflictFlag": 5,
  "current": null,
  "failed": Array [
    Object {
      "name": "index.html",
    },
  ],
  "files": Array [
    Object {
      "name": "favicon.ico",
    },
    Object {
      "name": "bg.jpg",
    },
    Object {
      "name": "logo.png",
    },
    Object {
      "name": "foo.gif",
    },
    Object {
      "name": "bar.png",
    },
    Object {
      "name": "baz.js",
    },
    Object {
      "name": "lorem.css",
    },
  ],
  "skipped": Array [],
}
`;

exports[`12- file skip action. Should add an item under "skipped".conflictFlag should be still 5 1`] = `
Object {
  "completed": Array [
    Object {
      "name": "style.css",
    },
    Object {
      "name": "main.js",
    },
    Object {
      "name": "vendor.js",
    },
    Object {
      "name": "manifest.json",
    },
  ],
  "conflict": false,
  "conflictFlag": 5,
  "current": null,
  "failed": Array [
    Object {
      "name": "index.html",
    },
  ],
  "files": Array [
    Object {
      "name": "bg.jpg",
    },
    Object {
      "name": "logo.png",
    },
    Object {
      "name": "foo.gif",
    },
    Object {
      "name": "bar.png",
    },
    Object {
      "name": "baz.js",
    },
    Object {
      "name": "lorem.css",
    },
  ],
  "skipped": Array [
    Object {
      "name": "favicon.ico",
    },
  ],
}
`;

exports[`13- file skip action. Should add an item under "skipped".conflictFlag should be still 5 1`] = `
Object {
  "completed": Array [
    Object {
      "name": "style.css",
    },
    Object {
      "name": "main.js",
    },
    Object {
      "name": "vendor.js",
    },
    Object {
      "name": "manifest.json",
    },
  ],
  "conflict": false,
  "conflictFlag": 5,
  "current": null,
  "failed": Array [
    Object {
      "name": "index.html",
    },
  ],
  "files": Array [
    Object {
      "name": "logo.png",
    },
    Object {
      "name": "foo.gif",
    },
    Object {
      "name": "bar.png",
    },
    Object {
      "name": "baz.js",
    },
    Object {
      "name": "lorem.css",
    },
  ],
  "skipped": Array [
    Object {
      "name": "favicon.ico",
    },
    Object {
      "name": "bg.jpg",
    },
  ],
}
`;
